mscore/events.cpp

```diff
@@ -849,7 +849,11 @@ void ScoreView::contextMenuEvent(QContextMenuEvent* ev)
       QPoint gp          = ev->globalPos();
       editData.startMove = toLogical(ev->pos());
       editData.buttons   = Qt::NoButton;
-      Element* e         = elementNear(editData.startMove);
+      Element* e         = nullptr;
+      if (ev->reason() == QContextMenuEvent::Keyboard)
+            e = score()->selection().element();
```

Comment:
Also wouldn't this make it impossible to open context menu for range or list selection? Maybe these cases should be handled too.

Reply:
Oh, because Selection::element() returns null in those cases?  Good point, here it wouldn't hurt to return *something*.  Although, it's much less important, there isn't much in those context menus that can't be accessed another way.  Maybe just split staff?  Anyhow, elementNear() won't help as the location data is not useful, but I'll see what makes sense.

Reply:
Well, at a closer look, this will just behave like it had behaved before, since [this branch](https://github.com/musescore/MuseScore/blob/7278beddfaf6bab151fbcc896e6ed01b185a6dce/mscore/events.cpp#L868-L881) of code would be executed. So there is no regression with this patch.

That branch could probably be improved though by trying to find a relevant measure with something like [`selection().findMeasure()`](https://github.com/musescore/MuseScore/blob/7278beddfaf6bab151fbcc896e6ed01b185a6dce/libmscore/select.h#L191) or [`selection().firstChordRest().findMeasure()`](https://github.com/musescore/MuseScore/blob/7278beddfaf6bab151fbcc896e6ed01b185a6dce/libmscore/select.h#L189) in case of keyboard-originated context menu event and, following the existing logic, showing measure menu or usual page menu depending on whether a relevant measure was found. Maybe it would be reasonable to add it to this PR as well? If not, then it can be merged as it is anyway.

Reply:
Not sure what kind of timetable you are looking at.  I can't guarantee I'd get to this today or tomorrow (although I might).  But I'd like to make the change in any case, it's a good improvement.  So whether this is merged now or later, I would want to do that.

Reply:
Actually the current plan is to finish merging changes for 3.3.3 release on Saturday, so if you add it until then it would perfectly fit into the current timeline. If not, then I would just merge this PR as it is, it is already an improvement anyway. So there is no hurry here in any case.